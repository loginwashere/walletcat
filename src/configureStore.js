import { createStore, applyMiddleware, compose } from 'redux'
import reducers from './reducers'
import middleware from './middleware'
import { addLocaleData } from 'react-intl'
import { DEFAULT_LOCALE } from './config'
import localeMessages from './translations'

import enLocaleData from 'react-intl/locale-data/en'
import ruLocaleData from 'react-intl/locale-data/ru'
import ukLocaleData from 'react-intl/locale-data/uk'

addLocaleData(enLocaleData)
addLocaleData(ruLocaleData)
addLocaleData(ukLocaleData)

const locale = DEFAULT_LOCALE
const messages = localeMessages[locale]

const initialState = {
  intl: {
    locale,
    messages,
  }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const configureStore = (preloadedState = {}) => createStore(
  reducers,
  {
    ...preloadedState,
    ...initialState
  },
  composeEnhancers(applyMiddleware(...middleware))
)

export default configureStore
