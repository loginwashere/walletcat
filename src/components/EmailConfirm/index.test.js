import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from '../../configureStore'
import { EmailConfirm } from '.'

describe('components:EmailConfirm:', () => {
  it('renders without crashing', () => {
    const store = configureStore()
    const dispatch = jest.fn()
    const code = 'test'
    const tree = renderer.create(
      <Provider store={store}>
        <EmailConfirm dispatch={dispatch}
                      isEmailConfirmed={false}
                      isEmailConfirmResent={false}
                      isFetching={false}
                      code={code} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
