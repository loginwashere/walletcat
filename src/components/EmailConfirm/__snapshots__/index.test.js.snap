exports[`components:EmailConfirm: renders without crashing 1`] = `
<div>
  <h1>
    Email confirm
  </h1>
  <div>
    <div
      className="alert alert-danger"
      role="alert">
      <p>
        <strong>
          That\'s weird!
        </strong>
         Something went wrong
      </p>
    </div>
    <form
      className="form-inline"
      onSubmit={[Function]}>
      <div
        className="row">
        <div
          className="col-sm-12 col-xs-12">
          <div
            className="form-group has-feedback">
            <h2>
              Request another confirmation email
            </h2>
            <div />
          </div>
        </div>
      </div>
      <div
        className="row">
        <div
          className="col-sm-3 col-xs-12">
          <div
            className="form-group has-feedback">
            <label
              className="sr-only control-label"
              htmlFor={undefined}>
              email@mail.com
            </label>
            <input
              autoFocus={undefined}
              className="form-control"
              id={undefined}
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="email@mail.com"
              required={undefined}
              type="email"
              value="" />
          </div>
        </div>
        <div
          className="col-sm-3 col-xs-12">
          <button
            className="btn btn-info"
            disabled={true}
            type="submit">
            Request another confirmation email
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
